recipe-client-addon/lib/FilterExpressions.jsm

```diff
@@ -41,3 +43,32 @@ this.FilterExpressions = {
     return jexl.eval(onelineExpr, context);
   },
 };
+
+/**
+ * Return an array of the given object's own keys (specifically, its enumerable
+ * properties), or undefined if the argument isn't an object.
+ * @param {Object} obj
+ * @return {Array[String]|undefined}
+ */
+function keys(obj) {
+  if (typeof obj !== "object" || obj === null) {
+    return undefined;
```

Comment:
[I chose undefined specifically](https://github.com/mozilla/normandy/pull/816#discussion_r123776562), and I think being explicit about that choice is best. Also, I think the linter complains when you have branches in a function that don't return an explicit value when other branches do, so `return;` would fail linting.